src/darker/config.py

```diff
@@ -114,23 +114,41 @@ def override_color_with_environment(pyproject_config: DarkerConfig) -> DarkerCon
     return config
 
 
-def load_config(srcs: Iterable[str]) -> DarkerConfig:
-    """Find and load Darker configuration from given path or pyproject.toml
+def load_config(path: Optional[str], srcs: Iterable[str]) -> DarkerConfig:
+    """Find and load Darker configuration from a TOML configuration file
 
-    :param srcs: File(s) and directory/directories which will be processed. Their paths
-                 are used to look for the ``pyproject.toml`` configuration file.
+    Darker determines the location for the configuration file by trying the following:
+    - the file path in the `path` argument, given using the ``-c``/``--config`` command
+      line option
+    - ``pyproject.toml`` inside the directory specified by the `path` argument
+    - ``pyproject.toml`` from a common parent directory to all items in `srcs`
+    - ``pyproject.toml`` in the current working directory if `srcs` is empty
+
+    :param path: The file or directory specified using the ``-c``/``--config`` command
+                 line option, or `None` if the option was omitted.
+    :param srcs: File(s) and directory/directories to be processed by Darker.
 
     """
-    path = find_project_root(tuple(srcs or ["."])) / "pyproject.toml"
-    if path.is_file():
-        pyproject_toml = toml.load(path)
-        config = cast(
-            DarkerConfig, pyproject_toml.get("tool", {}).get("darker", {}) or {}
-        )
-        replace_log_level_name(config)
-        validate_config_output_mode(config)
-        return config
-    return {}
+    if path:
+        for candidate_path in [Path(path), Path(path) / "pyproject.toml"]:
```

Comment:
```suggestion
        for candidate_path in [Path(path), Path(path, "pyproject.toml")]:
```
I think is the better way to construct the path object.